src/app/search/loadouts/search-filters/freeform.ts

```diff
@@ -22,6 +44,49 @@ const freeformFilters: FilterDefinition<
       return (loadout) => test(loadout.name);
     },
   },
+  {
+    keywords: ['subclass'],
+    description: tl('LoadoutFilter.Subclass'),
+    format: 'freeform',
+    suggestionsGenerator: ({ loadouts, d2Definitions }) => {
+      if (!loadouts || !d2Definitions) {
+        return [];
+      }
+      const damageDefs = damageDefsByDamageType(d2Definitions);
+      // TODO (ryan) filter on currently selected character. This info is currently localized
+      // to the page, so we need to lift that up before it can be done.
+      return deduplicate(
+        loadouts.flatMap((loadout) => {
+          const subclass = subclassDefFromLoadout(loadout, d2Definitions);
+          const damageType = subclass ? getDamageTypeForSubclassDef(subclass) : undefined;
+          // DamageType.None is 0
+          const damageName =
+            damageType !== undefined ? damageDefs[damageType]?.displayProperties.name : undefined;
+          return [
+            subclass
+              ? `subclass:${quoteFilterString(subclass.displayProperties.name.toLowerCase())}`
+              : undefined,
+            damageName ? `subclass:${quoteFilterString(damageName.toLowerCase())}` : undefined,
```

Comment:
Interestingly, these will be localized. I think that's good?

Reply:
Yeah I was thinking about that. I think that brings some awkwardness with the filters which aren't localised but overall being closer to the in game display is probably good.